RecoHI/HiJetAlgos/plugins/GenericVoronoiAlgorithm.cc

```diff
@@ -0,0 +1,994 @@
+						index_row, index_column_block, 1.0 / weight);
+
+					for (std::vector<size_t>::const_iterator
+							 iterator_recombine_unsigned_inner =
+							 iterator_recombine_unsigned_outer->begin();
+						 iterator_recombine_unsigned_inner !=
+							 iterator_recombine_unsigned_outer->end();
+						 iterator_recombine_unsigned_inner++) {
+						if (_event[*iterator_recombine_unsigned_inner].momentum_perp_subtracted >= 0) {
+							const size_t index_column_cost =
+								nblock +
+								positive_index[*iterator_recombine_unsigned_inner];
+
+							p->push_back_coefficient(
+								index_row, index_column_cost, 1);
+							index_column_max =
+								std::max(index_column_max, index_column_cost);
+						}
+					}
+					index_row++;
+
+					p->push_back_row(
+						bpmpd_problem_t::greater_equal,
+						-sum_unequalized);
+
+					p->push_back_coefficient(
+						index_row, index_column_block, _positive_bound_scale / weight);
+
+					for (std::vector<size_t>::const_iterator iterator_recombine_unsigned_inner = iterator_recombine_unsigned_outer->begin();
+						 iterator_recombine_unsigned_inner != iterator_recombine_unsigned_outer->end();
+						 iterator_recombine_unsigned_inner++) {
+						if (_event[*iterator_recombine_unsigned_inner].momentum_perp_subtracted >= 0) {
+							const size_t index_column_cost =
+								nblock +
+								positive_index[*iterator_recombine_unsigned_inner];
+
+							p->push_back_coefficient(
+								index_row, index_column_cost, -1);
+							index_column_max =
+								std::max(index_column_max, index_column_cost);
+						}
+					}
+					index_row++;
+				}
+
+				}
+
+			}
+		}
+	}
+
+	// Epsilon that breaks the degeneracy, in the same units
+	// as the pT of the event (i.e. GeV)
+	static const double epsilon_degeneracy = 1e-2;
+
+	// Columns (variables and the objective coefficients) of
+	// the LP problem
+	//
+	// Cost variables (objective coefficient 1)
+	for (size_t i = 0; i < nsuperblock; i++) {
+		p->push_back_column(
+			1, 0, bpmpd_problem_t::infinity);
+	}
+	for (size_t i = nsuperblock; i < nsuperblock + nstaggered_block; i++) {
+		p->push_back_column(
+			0, 0, bpmpd_problem_t::infinity);
+	}
+	for (size_t i = nsuperblock + nstaggered_block; i < nsuperblock + 2 * nstaggered_block; i++) {
+		p->push_back_column(
+			0, 0, bpmpd_problem_t::infinity);
+	}
+	for (size_t i = nsuperblock + 2 * nstaggered_block; i < _ncost; i++) {
+		p->push_back_column(
+			0, 0, bpmpd_problem_t::infinity);
+	}
+	//fprintf(stderr, "%s:%d: %lu %lu\n", __FILE__, __LINE__, index_column_max, recombine_tie.size());
+	// Energy transfer coefficients t_kl (objective
+	// coefficient 0 + epsilon)
+	for (size_t i = _ncost; i <= index_column_max; i++) {
+		p->push_back_column(
+			epsilon_degeneracy * _recombine_tie[i - _ncost],
+			0, bpmpd_problem_t::infinity);
+	}
+}
+void GenericVoronoiAlgorithm::equalize(void)
+{
+	bpmpd_problem_t lp_problem = reinterpret_cast<bpmpd_environment_t *>(_lp_environment)->problem();
```

Comment:
This is left over and removed together with the class (the previous class wrapper simply calls the default constructor). Fixed in the commit below.